cmd/shim/main.go

```diff
@@ -286,22 +286,30 @@ func shim() int {
 	outWriter := io.MultiWriter(stdoutFile, os.Stdout)
 	errWriter := io.MultiWriter(stderrFile, os.Stderr)
 
-	// Only initialize scrubbing logic if secrets are set:
 	if len(secretsToScrub.Envs) == 0 && len(secretsToScrub.Files) == 0 {
 		cmd.Stdout = outWriter
 		cmd.Stderr = errWriter
 	} else {
 		envToScrub := os.Environ()
-		scrubOutWriter, err := NewSecretScrubWriter(outWriter, currentDirPath, shimFS, envToScrub, secretsToScrub)
+		stdoutPipe, err := cmd.StdoutPipe()
```

Comment:
I pushed the same changes and didn't see the issue again however I'm facing the following scenarios when testing latest `main`:
- On M1 with Docker Desktop all `TestContainerWithMountedSecretOwner` work fine.
- On my Linux box, if I use a regular user I see this issue:
```
=== RUN   TestContainerWithMountedSecretOwner/userid
#1 resolve image config for docker.io/library/alpine:3.16.2
    container_test.go:3671: 
        	Error Trace:	/home/matias/go/src/github.com/dagger/dagger/core/integration/container_test.go:3671
        	Error:      	Not equal: 
        	            	expected: "auser auser"
        	            	actual  : "auser "
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-auser auser
        	            	+auser 
        	Test:       	TestContainerWithMountedSecretOwner/userid
```
- If I ran everything as root (`sudo ./hack/dev`, etc.) tests pass.

I tried the same with this branch, when running with `sudo` test pass, regular usar fails.

So I'm wandering the following:
- I see there's some logic around inheriting permissions in 48fb31df3c383da743c9cba5723f4291e0e6f8df. Could this break depending on which user is being used in the host?
- Based on the above, do we run tests in some privileged mode? And/or did we change anything related to it since the last failing test report you did?

Could be good to involve @vito here too, thanks! Besides that I don't see that the previously failing test is related to my changes but would be good to understand why the issue happened.

Reply:
@gerhard Interesting, for me it works properly on Mac using a regular user.
The issue seems to occur on Linux only.

Reply:
I would also guess this is some kind of race condition or issue with the secret scrubbing. These tests should have no relation to what user runs them on the host. 🤔 I can't think of a way to convince `stat -c '%U %G'` to print an empty `%G`, so it seems more like the last chunk of output is bring dropped. Happy to be wrong, but that's my guess.

Reply:
@vito makes sense however I also see it in latest `main` -without my changes-. Will debug a bit more on Monday.

Reply:
I've verified this again, tried the following:
- Ran `TestContainerWithMountedSecretOwner` multiple times in both branches (`main` and this one), with and without race. Didn't see any issues so far.
- Added a `panic()` to the block where secret scrub logic is initialized [here](https://github.com/dagger/dagger/pull/5149/files#diff-e4db2b909335ff595a2cd89bad7531692d487f29f9cdc57a744b87f6f86e4e77R292) so that the execution breaks If this part of the code is reached: regular secret scrub integration tests like `TestBigSecretScrubbed` got the `panic` but `TestContainerWithMountedSecretOwner` didn't, this indicates that the logic that's part of this PR isn't involved in `TestContainerWithMountedSecretOwner` at all.

Reply:
Also the reason why test `TestContainerWithMountedSecretOwner/userid` has been failing to me seems to be a caching issue. Even with rebuilds it would still print the wrong output (`a user `).
I renamed the test to `TestContainerWithMountedSecretOwner/userid-111`, ran `go test -v -count=1 -timeout 30s -run ^TestContainerWithMountedSecretOwner/userid-111$ github.com/dagger/dagger/core/integration` and the results are just fine!

Reply:
Are you saying that if `TestContainerWithMountedSecretOwner/userid` test changes, the previous run remains cached? That would be surprising behaviour which we should address.

Reply:
I continued running this as mentioned, and even though I have hit other issues - still to create them - I was not able to reproduce the original failure. I have seen it fail **once** locally, before I started looking for it, but not since.